lib/ansible/module_utils/vmware.py

```diff
@@ -29,20 +30,41 @@ class TaskError(Exception):
     pass
 
 
-def wait_for_task(task):
+def wait_for_task(task, max_backoff=64, timeout=3600):
+    """Function to wait for given task using exponential back-off algorithm.
+
+    Args:
+        task: VMware task object
+        max_backoff: Maximum amount of sleep time in seconds
+        timeout: Timeout for the given task in seconds
+
+    Returns: TaskError for failure or True with result for successful task.
+    """
+    failure_counter = 0
+    start_time = time.time()
+
+    # List of generic errors
+    gen_errors = (
+        vim.fault.CannotAccessVmConfig,
+        vmodl.fault.InvalidArgument,
+    )
 
     while True:
+        if time.time() - start_time >= timeout:
+            raise TaskError("Timeout")
         if task.info.state == vim.TaskInfo.State.success:
             return True, task.info.result
```

Comment:
This is a bad API: you indicate failure/success by both raising exceptions and adding metadata via return value.
It's common in C or Go to use return status codes, but in Python it's just fine to raise exceptions for failures and then return ready-to-use values for successful outcome. It's pointless to always return static `True` additionally to useful payload.

I suggest you to simplify this by wiping out this boolean part and returning just useful result, not a tuple.

Reply:
I totally agree with your comments, but like I said earlier - this is a legacy function and used by lot of modules. I will need to go and change every module which uses it and test it again. Changing might have detrimental effects rather than doing something good. Let me know if your views on this.

Reply:
Okay, if it's used by lot's of modules it should go to the other PR.